addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/handler/KonnectedHandler.java

```diff
@@ -0,0 +1,345 @@
+        logger.debug("The callback ip address is: {}", callbackIpAddress);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // get the zone number in integer form
+        Integer zone = Integer.parseInt(channelUID.getId().substring((channelUID.getId().length() - 1)));
+        logger.debug("The channelUID is: {} and the zone is :", channelUID.getAsString(), zone);
+        // convert the zone to the pin based on value at index of zone
+        Integer pin = Arrays.asList(PIN_TO_ZONE).get(zone);
+        String scommand = command.toString();
+        if (scommand.equalsIgnoreCase("ON")) {
+            sendActuatorCommand("1", pin, channelUID);
+            logger.trace("The command being sent to pin {} for channel:  is 1", channelUID.getAsString(), pin);
+        } else if (scommand.equalsIgnoreCase("OFF")) {
+            sendActuatorCommand("0", pin, channelUID);
+            logger.trace("The command being sent to pin {} for channel:  is 0", channelUID.getAsString(), pin);
+        } else if ((scommand.equalsIgnoreCase("OPEN")) || (scommand.equalsIgnoreCase("CLOSED"))) {
+            logger.debug("A command was sent to a sensor type so we are ignoring the command");
+        }
+    }
+
+    /**
+     * Process a {@link WebHookEvent that has been received by the Servlet from a Konnected module
+     *
+     * @param pin     the pin number which which was activated
+     * @param state   the state of the pin
+     * @param setAuth the token sent with the response
+     */
+    public void handleWebHookEvent(String pin, String State, String sentAuth) {
+        // get the zone number based off of the index location of the pin value
+        String sentZone = Integer.toString(Arrays.asList(PIN_TO_ZONE).indexOf(Integer.parseInt(pin)));
+        // check that the zone number is in one of the channelUID definitions
+        getThing().getChannels().forEach(channel -> {
+            ChannelUID channelId = channel.getUID();
+            String zoneNumber = channelId.getId().substring((channelId.getId().length() - 1));
+            // if the string zone that was sent equals the last digit of the channelId found process it as the
+            // channelId else do nothing
+            if (sentZone.equalsIgnoreCase(zoneNumber)) {
+                logger.debug("The channelID of the event is: {}, the Auth Token is: {}", channelId, sentAuth);
+                // check that the auth token sent matches the authToken configured on the Thing
+                if (sentAuth.endsWith(config.authToken)) {
+                    String itemType = channel.getAcceptedItemType();
+                    // check if the itemType has been defined for the zone received
+                    if (itemType == null) {
+                        logger.error(
+                                "The itemType is not configured for channel {} on thing {}, please get your configuration.",
```

Comment:
so that is what i assumed as well, however without this check eclipse give me a potential null pointer access warning.  So that's why i put the check in.  I guess its not needed and i can safely ignore the warning? But i thought it better to have the check in and log the error?